include/boost/http_proto/service/zlib_service.hpp

```diff
@@ -47,10 +48,14 @@ struct deflate_decoder_service
 
     virtual
     filter&
-    make_filter(detail::workspace& ws) const = 0;
-};
+    make_deflate_filter(
+        http_proto::detail::workspace& ws) const = 0;
```

Comment:
should `deflate_decoder_service` be private?

Reply:
My understanding is that it has to be public so that users can setup zlib via the `context`, i.e. this is how they opt into requiring zlib for their applications.

Reply:
Users only need one public free function to install the service into the context.  The API for creating instances of the filters is only used by the implementation. I could be wrong...you will need to check.